src/MergeContext.js

```diff
@@ -1063,14 +1248,39 @@ class PullRequest {
         const waitForMergeable = !(this._stagedPosition && this._stagedPosition.merged());
         const pr = await GH.getPR(this._prNumber(), waitForMergeable);
         assert(pr.number === this._prNumber());
+
         this._rawPr = pr;
+
+        if (this._prMergeable()) {
+            const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+            this._mergeCommit = await GH.getCommit(mergeSha);
+            try {
+                this._commitMessage = new CommitMessage(this._rawPr, this._mergeCommit.author);
+            } catch (e) {
+                this._logEx(e, "cannot parse commit message");
+            }
+        }
     }
 
     // Whether the commit message configuration remained intact since staging.
-    async _messageIsFresh() {
-        const result = this._prMessage() === this._stagedCommit.message;
+    _messageIsFresh() {
```

Comment:
Since we are checking both message text and author fields, I suggest replacing "commit message configuration" with "staged commit metadata" and renaming _messageIsFresh() to _stagedCommitMetadataIsFresh().